pkg/keycloak/keycloak.go

```diff
@@ -157,10 +162,15 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 
 			labelSelector := fmt.Sprintf("serviceInstanceID=%s,serviceType=%s", kc.Spec.InstanceUID, "keycloak")
 			secretList, err := h.k8sClient.CoreV1().Secrets(kc.Namespace).List(metav1.ListOptions{LabelSelector: labelSelector})
-			if err != nil || len(secretList.Items) == 0 {
+			if err != nil {
 				return errors.Wrap(err, "error reading admin credentials")
 			}
 
+			if len(secretList.Items) == 0 {
+				logrus.Debug("keycloak service credentials not found")
+				return nil
+			}
```

Comment:
Separated the check here as it wasn't obvious what was going on previously since "error.Wrap" returns nil if the "err" object it is supposedly wrapping is nil. Deliberately not returning an error here as this is expected during a provision that this will be empty until it's completed the provision of KeyCloak.